SPECS/golang/golang.spec

```diff
@@ -56,19 +60,44 @@ popd
 mv -v %{_topdir}/BUILD/go-bootstrap %{_libdir}/golang
 export GOROOT=%{_libdir}/golang
 
-# Use go1.4 bootstrap to compile go%{bootstrap_compiler_version} (bootstrap)
+# Use go1.4 bootstrap to compile go%{bootstrap_compiler_version_0}
 export GOROOT_BOOTSTRAP=%{_libdir}/golang
-mkdir -p %{_topdir}/BUILD/go%{bootstrap_compiler_version}
-tar xf %{SOURCE2} -C %{_topdir}/BUILD/go%{bootstrap_compiler_version} --strip-components=1
-pushd %{_topdir}/BUILD/go%{bootstrap_compiler_version}/src
+mkdir -p %{_topdir}/BUILD/go%{bootstrap_compiler_version_0}
+tar xf %{SOURCE2} -C %{_topdir}/BUILD/go%{bootstrap_compiler_version_0} --strip-components=1
+pushd %{_topdir}/BUILD/go%{bootstrap_compiler_version_0}/src
 CGO_ENABLED=0 ./make.bash
 popd
+# Nuke the older %{bootstrap_compiler_version_0}
+rm -rf %{_libdir}/golang
+mv -v %{_topdir}/BUILD/go%{bootstrap_compiler_version_0} %{_libdir}/golang
+export GOROOT=%{_libdir}/golang
+
+
+# Use go%{bootstrap_compiler_version_0} bootstrap to compile go%{bootstrap_compiler_version_1} (bootstrap)
+export GOROOT_BOOTSTRAP=%{_libdir}/golang
+mkdir -p %{_topdir}/BUILD/go%{bootstrap_compiler_version_1}
+tar xf %{SOURCE3} -C %{_topdir}/BUILD/go%{bootstrap_compiler_version_1} --strip-components=1
+pushd %{_topdir}/BUILD/go%{bootstrap_compiler_version_1}/src
+CGO_ENABLED=0 ./make.bash
+popd
+# Nuke the older %{bootstrap_compiler_version_1}
+rm -rf %{_libdir}/golang
+mv -v %{_topdir}/BUILD/go%{bootstrap_compiler_version_1} %{_libdir}/golang
+export GOROOT=%{_libdir}/golang
 
-# Nuke the older go1.4 bootstrap
+# Use %{bootstrap_compiler_version_1} to compile %{bootstrap_compiler_version_2}
+export GOROOT_BOOTSTRAP=%{_libdir}/golang
+mkdir -p %{_topdir}/BUILD/go%{bootstrap_compiler_version_2}
+tar xf %{SOURCE4} -C %{_topdir}/BUILD/go%{bootstrap_compiler_version_2} --strip-components=1
+pushd %{_topdir}/BUILD/go%{bootstrap_compiler_version_2}/src
+CGO_ENABLED=0 ./make.bash
+popd
+
+# Nuke the older %{bootstrap_compiler_version_1}
 rm -rf %{_libdir}/golang
 
-# Make go%{bootstrap_compiler_version} as the new bootstrapper
-mv -v %{_topdir}/BUILD/go1.19.12 %{_libdir}/golang
+# Make go%{bootstrap_compiler_version_2} as the new bootstrapper
+mv -v %{_topdir}/BUILD/go1.22.4 %{_libdir}/golang
```

Comment:
I don't understand. It looks like 22.4 is the final version we're releasing with this spec, so why use it as a bootstrap step?

Reply:
To avoid confusion, I have renamed bootstrap_compiler_version_2 to go_version_to_build.